src/server.c

```diff
@@ -290,6 +290,18 @@ int dictSdsKeyCompare(dict *d, const void *key1, const void *key2) {
     return memcmp(key1, key2, l1) == 0;
 }
 
+size_t dictSdsKeyLen(const void *key) { 
+    return sdsAllocSize((sds)key);
+}
+
+size_t dictSdsKeyToBytes(unsigned char *buf, const void *key, unsigned char *header_size) {
+    sds keysds = (sds)key;
+    size_t n_bytes = sdsAllocSize(keysds);
```

Comment:
[Readability suggestion] This size computation logic needs to match `dictSdsKeyToBytes` to avoid buffer overrun bugs.

```suggestion
    size_t n_bytes = dictSdsKeyLen(keysds);
```

 Or better yet, let the caller provide the buffer length and we can assert on it being larger than `dictSdsKeyLen(key)`
 
 ```c
 size_t dictSdsKeyCopy(unsigned char *buf, size_t buf_len, const void *key, unsigned char *key_offset)
 ```